biogeochem/EDLoggingMortalityMod.F90

```diff
@@ -791,4 +811,74 @@ subroutine logging_litter_fluxes(currentSite, currentPatch, newPatch, patch_site
       return
    end subroutine logging_litter_fluxes
 
+  ! =====================================================================================
+
+   subroutine UpdateHarvestC(currentSite,bc_out)
+
+      ! ----------------------------------------------------------------------------------
+      ! Added by Shijie Shu.
+      ! This subroutine is called when logging is completed and need to update 
+      ! Harvested C flux in HLM.
+      ! ----------------------------------------------------------------------------------
+      use EDtypesMod             , only : ed_site_type
+      use EDTypesMod           , only : AREA_INV
+      use PRTGenericMod          , only : element_pos
+      use PRTGenericMod          , only : carbon12_element
+      use FatesInterfaceTypesMod , only : bc_out_type
+      use EDParamsMod            , only : pprodharv10_forest_mean
+  
+      ! Arguments
+      type(ed_site_type), intent(inout), target :: currentSite     ! site structure
+      type(bc_out_type), intent(inout)          :: bc_out
+  
+      integer :: icode
+      real(r8) :: unit_trans_factor
+  
+
+      ! Flush the older value before update
+      if(logging_time) then
+         bc_out%hrv_deadstemc_to_prod10c = 0._r8
+         bc_out%hrv_deadstemc_to_prod100c = 0._r8
+      end if
+  
+      ! First test tropic forest (PFT=1)
+      ! Calculate the unit transfer factor (from kgC m-2 day-1 to gC m-2 s-1)
+      ! for icode == 2, icode < 0, and icode > 10000 is one time harvest, thus
+      ! shall distribute into everyday
```

Comment:
@glemieux just brought up this point with me:  @sshu88 are we sure that we need different conversion factors for different event codes?  The wood_product is a rate, it isn't time integrated over the month.  It seems that the conversion factor should be 1000 * days_per_sec for all cases.  Can you clarify please?

Reply:
sorry, I still don't think I understand.  It seems to me that the wood_product rate will always have the same units, regardless of how many times logging occurs per month.  Appologies if I'm missing something obvious here.

Reply:
Sorry for the confusion. I discussed this issue with @jenniferholm and it seems like my code comments misled users. 

`wood_product` is in the unit of kgC m-2 day-1 and represent the harvested C flux, but not yet the input fluxes into the bookkeeping model in ELM (`bc_out%hrv_deadstemc_to_prod10c` and `bc_out%hrv_deadstemc_to_prod100c`). ELM LULCC perform logging once a year and distribute the harvested C into s-1 rate, i.e., a constant harvest rate for each model timestep. I added this piece of code to **distribute the harvested C flux from one day to a constant s-1 rate for each model timestep until a new logging event happen** for different type of logging event code (once a year, per month or everyday, etc.), thus to keep consistent to the behavior of ELM. 

Here attaches a figure showing the difference of the size of total product C pools between 1) the case directly putting all logged C into product pool ("spike") and 2) the case distributing the logged C to each day across the year then send C to product pool ("uniform case"):
![comparison_uniform_vs_spike](https://user-images.githubusercontent.com/92333861/158904485-9c86e1e3-4a24-4606-a782-5eb4c23f10af.png)

Since this is a ELM specific feature, and also can be achieved by changing the logging event code in FATES parameter file, I will follow your suggestion to revise the code.

@jenniferholm, let me know if I have missed something.

Thank you for checking out this issue.

Reply:
Thanks @sshu88 now I understand

Reply:
Hi Shijie - thanks for doing these two comparison tests. 1) putting the logged C into product pool within one day, and 2) averaging the C product pool over the whole year, which I believe you said is what ELM prefers to use, right?   Could you please clarify for me what the decision is for the FATES code?  
Are you going to have the harvested C in FATES be HLM agnostic, and put all the harvest C into the product pool in a single time step (spike case), then on the ELM side do the unit transfer to constant s-1 rate?  Sorry if you are repeating yourself, I just wanted to double check.

Reply:
@jenniferholm Thank you for your comment. 

I decided to add these codes to avoid spurious response from ELM bookkeeping model if we provide "spiky" C input to ELM bookkeeping model, thus the simulation results shall look like what "uniform" case showed no matter what logging event code we provide.  But now I have tested both cases and the ELM bookkeeping model can handle "spiky" inputs, although the results varied due to a different timing of inputs. I think we can remove these codes and use the FATES logging event code to control how "spiky" the C input to product pool. Any further suggestions?